packages/wrangler/src/api/startDevWorker/RemoteRuntimeController.ts

```diff
@@ -1,30 +1,303 @@
+				) {
+					logger.error("Error on remote worker:", err);
+				}
+			} else {
+				throw err;
+			}
+		}
+	}
+
+	async #onBundleComplete({ config, bundle }: BundleCompleteEvent, id: number) {
+		try {
+			const routes = config.triggers
+				?.filter(
+					(trigger): trigger is Extract<Trigger, { type: "route" }> =>
+						trigger.type === "route"
+				)
+				.map((trigger) => {
+					const { type: _, ...route } = trigger;
+					if (!("custom_domain" in route)) {
+						return route.pattern;
+					}
+					return route;
+				});
+
+			if (!config.dev?.auth) {
+				throw new MissingConfigError("config.dev.auth");
+			}
+			const auth = await unwrapHook(config.dev.auth);
+
+			if (this.#session) {
+				logger.log(chalk.dim("⎔ Detected changes, restarted server."));
+			}
+
+			this.#session ??= await this.#previewSession({
+				accountId: auth.accountId,
+				env: config.env, // deprecated service environments -- just pass it through for now
+				legacyEnv: config.legacyEnv, // wrangler environment -- just pass it through for now
+				host: config.dev.origin?.hostname,
+				routes,
+				sendMetrics: config.sendMetrics,
+			});
+
+			const bindings = (
+				await convertBindingsToCfWorkerInitBindings(config.bindings)
+			).bindings;
+
+			const token = await this.#previewToken({
+				bundle,
+				modules: bundle.modules,
+				accountId: auth.accountId,
+				name: config.name,
+				legacyEnv: config.legacyEnv,
+				env: config.env,
+				isWorkersSite: config.site !== undefined,
+				assetPaths: config.site?.path
+					? {
+							baseDirectory: config.site.path,
+							assetDirectory: "",
+							excludePatterns: config.site.exclude ?? [],
+							includePatterns: config.site.include ?? [],
+						}
+					: undefined,
+				format: bundle.entry.format,
+				// TODO: Remove this passthrough
+				bindings: config._bindings ? config._bindings : bindings,
+				compatibilityDate: config.compatibilityDate,
+				compatibilityFlags: config.compatibilityFlags,
+				usageModel: config.usageModel,
+				routes,
+			});
+
+			// If we received a new `bundleComplete` event before we were able to
+			// dispatch a `reloadComplete` for this bundle, ignore this bundle.
+			// If `token` is undefined, we've surfaced a relevant error to the user above, so ignore this bundle
+			if (id !== this.#currentBundleId || !token) {
+				return;
+			}
+
+			const accessToken = await getAccessToken(token.host);
+
+			this.emitReloadCompleteEvent({
+				type: "reloadComplete",
+				bundle,
+				config,
+				proxyData: {
+					userWorkerUrl: {
+						protocol: "https:",
+						hostname: token.host,
+						port: "443",
+					},
+					userWorkerInspectorUrl: {
+						protocol: token.inspectorUrl.protocol,
+						hostname: token.inspectorUrl.hostname,
+						port: token.inspectorUrl.port.toString(),
+						pathname: token.inspectorUrl.pathname,
+					},
+					userWorkerInnerUrlOverrides: {
+						hostname: token.host,
+					},
```

Comment:
This can't actually be used in remote-mode – the hostname is going to be `token.host` anyway